### PR TITLE
Share common code between contract

### LINKED ISSUE #42: Reorganize common contract code in util package
There are a lot of common structures and functions in contracts such as:
- `ballot struct`,
- `vote()`,
- `removeVote()`,
- `invokeID()`,
- `innerRingInvoker()`,
- ...
- many more.

They can be placed in separate package and it should be compiled with neo-go. This will heavily reduce complexity and size of the contracts.

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #42.

- [x] `type ballot`
- [x] `func vote`
- [x] `func removeVote`
- [x] `func invokeID`
- [x] `func innerRingInvoker`
- [x] `func setSerialized`
- [x] `func bytesEqual`

### CODE REVIEW

File: common/ballot.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package common
+
+type Ballot struct {
+	// ID of the voting decision.
+	ID []byte
+
+	// Number already voted inner ring nodes.
```
Comment (alexvanin) [inline common/ballot.go:7]: `Public keys of already voted inner ring nodes` will be more accurate.

Comment (cthulhu-rider) [reply common/ballot.go]: Fixed

File: common/ballot.go [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package common
+
+type Ballot struct {
```
Comment (alexvanin) [inline common/ballot.go:3]: I think ballot type can be defined in `vote.go`, it has not use outside of voting.

File: common/invoke.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package common
+
+import (
+	"github.com/nspcc-dev/neo-go/pkg/interop/crypto"
+	"github.com/nspcc-dev/neo-go/pkg/interop/runtime"
+)
+
+func InvokeID(args []interface{}, prefix []byte) []byte {
```
Comment (alexvanin) [inline common/invoke.go:8]: `InvokeID` used to identify txs and collect votes, so it better to move to `vote.go`.

File: common/invoke.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+	for i := range args {
+		arg := args[i].([]byte)
+		prefix = append(prefix, arg...)
+	}
+
+	return crypto.SHA256(prefix)
+}
+
+type IRNode struct {
```
Comment (alexvanin) [inline common/invoke.go:17]: Let's move inner ring list related code to `ir.go`: both `IRNode` struct and `InnerRingInvoker` function.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**